pkgs/build-support/rust/README.md

```diff
@@ -0,0 +1,39 @@
+# Updated fetchcargo behavior
+
+Changes to the `fetchcargo.nix` behavior that cause changes to the `cargoSha256`
+are somewhat disruptive, so historically we've added conditionals to provide
+backwards compatibility. We've now accumulated enough of these that it makes
+sense to do a clean sweep updating hashes, and delete the conditionals in the
+fetcher to simplify maintenance and implementation complexity. These are:
+
+1. When cargo vendors dependencies, it generates a config. Previously, we were
+   hard-coding our own config, but this fails if there are git dependencies. We
+   have conditional logic to sometimes copy the vendored cargo config in, and
+   sometimes not.
+
+2. When a user updates the src package, they may forget to update the
+   `cargoSha256`. We have a conditional flag to add the `Cargo.lock` into the
+   vendor dir for inspection, but it defaults to false.
+
+3. We were previously vendoring into a directory, but would like to vendor into
+   a compressed tar.gz file instead.
+
+
+## Migration plan
```

Comment:
I have updated the hash on every package that has `verifyCargoDeps = true;` in this PR, and verified that other rust packages with this set to false are not impacted, e.g., changing 1 character on the hash for `ripgrep` and re-building it produces an error that shows the old hash.  

I have a script to do the rest of them and set it to true everywhere, but figured I'd open the PR first for feedback about whether others agree this is a good idea in general first.  If so, once it's merged I can send the treewide follow-up PR that changes all the hashes and sets the attribute to true, then once that's merged I'll send another that removes the boolean.